backend/check_stac_metadata/utils.py

```diff
@@ -161,6 +145,20 @@ def save(self, key: str) -> None:
                 multihash=asset["multihash"],
             ).save()
 
+    def validate_directory(self, url: str, parent_metadata_url: str) -> None:
+        root_path = get_url_before_filename(self.traversed_urls[0])
+        if root_path != get_url_before_filename(url):
+            self.validation_result_factory.save(
+                url,
+                Check.MULTIPLE_DIRECTORIES,
+                ValidationResult.FAILED,
+                details={
+                    "message": f"“metadata file: {parent_metadata_url} links to {url}”"
+                    f" which exists in a different directory to the root "
+                    f"metadata file directory: “{root_path}”"
+                },
+            )
+
 
 def get_url_before_filename(url: str) -> str:
```

Comment:
We should `@lru_cache` this endpoint since it now gets run many times for the root URL.